File: MusicWeb.Server/MusicWeb.Models/Entities/Artists/Artist.cs
using MusicWeb.Models.Entities.Artists;
using MusicWeb.Models.Entities.Base;
using MusicWeb.Models.Entities.Origins;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicWeb.Models.Entities
{
    public class Artist : BaseEntity
    {
        public string Name { get; set; }
        public DateTime EstablishmentDate { get; set; }
        public string Bio { get; set; }
        public bool IsIndividual { get; set; }
        public bool IsBand { get; set; }

        public int? BandId { get; set; }
        public int CountryId{ get; set; }
        public int StateId{ get; set; }
        public int CityId{ get; set; }
        public virtual BandMember BandMember { get; set; }
        public virtual Country Country{ get; set; }
        public virtual State State{ get; set; }
        public virtual City City{ get; set; }

        public virtual ICollection<Album> Albums{ get; set; }
        public virtual ICollection<Song> Songs{ get; set; }
        public virtual ICollection<SongGuestArtist> SongGuestArtists{ get; set; }
        public virtual ICollection<ArtistComment> ArtistComments{ get; set; }
        public virtual ICollection<ArtistGenre> ArtistGenres{ get; set; }
        public virtual ICollection<UserFavoriteArtist> UserFavoriteArtists{ get; set; }
        public virtual ICollection<UserObservedArtist> UserObservedArtists{ get; set; }
        public virtual ICollection<ArtistsOnTheAlbum> ArtistsOnTheAlbums{ get; set; }
        public virtual ICollection<BandMember> Band { get; set; }

    }
}

File: MusicWeb.Client/src/models/Artist.js
export default class Artist {
  id = null;
  imagePath = "";
  name = "";
  establishmentDate = "";
  description = "";
  imagePath = "";
  isIndividual = false;
  isBand = false;
  bandId = 0;
  country = "";
  state = "";
  city = "";
  albums = [];
  songs = [];
  members = [];
  genres = [];
  rating = "";
  ratingsCount = "";
  favoriteCount = null;
  observedCount = null;

  constructor($data) {
    if ($data) {
      this.id = $data.id;
      this.imagePath = $data.imagePath;
      this.name = $data.name;
      this.establishmentDate = $data.establishmentDate;
      this.description = $data.bio;
      this.imagePath = $data.imagePath;
      this.isIndividual = $data.isIndividual ? $data.isIndividual : "";
      this.isBand = $data.isBand ? $data.isBand : "";
      this.bandId = $data.bandId;
      this.country = $data.country ? $data.country : "";
      this.state = $data.state ? $data.state : "";
      this.city = $data.city ? $data.city : "";
      this.albums = $data.albums ? $data.albums : [];
      this.songs = $data.songs ? $data.songs : [];
      this.members = $data.members
        ? $data.members.map((member) => member.name)
        : [];
      this.genres = $data.genres ? $data.genres.map((genre) => genre.name) : [];
      this.rating = $data.rating ? $data.rating : 0;
      this.ratingsCount = $data.ratingsCount ? $data.ratingsCount : 0;
      this.favoriteCount = $data.favoriteCount ? $data.favoriteCount : 0;
      this.observedCount = $data.observedCount ? $data.observedCount : 0;
    }
  }
}
